File: 0226-invert-binary-tree/0226-invert-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* invertTree(struct TreeNode* root){

    if (root == NULL){
        return root;
    }
    struct TreeNode* left_side;
    struct TreeNode* right_side;

    left_side = invertTree(root->left);
    right_side = invertTree(root->right);

    root->left = right_side;
    root->right = left_side;

    return (root);
}

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if not root:
            return root

        left_side = self.invertTree(root.left)
        right_side= self.invertTree(root.right)

        root.left, root.right = (right_side, left_side)
        return root
        